src/NzbDrone.Core/Notifications/Ntfy/NtfySettings.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using FluentValidation;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Notifications.Ntfy
+{
+    public class NtfySettingsValidator : AbstractValidator<NtfySettings>
+    {
+        public NtfySettingsValidator()
+        {
+            RuleFor(c => c.Topics).NotEmpty();
+            RuleFor(c => c.Priority).InclusiveBetween(1, 5);
+            RuleFor(c => c.ServerUrl).IsValidUrl().When(c => !c.ServerUrl.IsNullOrWhiteSpace());
+            RuleFor(c => c.ClickUrl).IsValidUrl().When(c => !c.ClickUrl.IsNullOrWhiteSpace());
+            RuleFor(c => c.UserName).NotEmpty().When(c => !c.Password.IsNullOrWhiteSpace());
+            RuleFor(c => c.Password).NotEmpty().When(c => !c.UserName.IsNullOrWhiteSpace());
+            RuleForEach(c => c.Topics).NotEmpty().Matches("[a-zA-Z0-9_-]+").Must(c => !InvalidTopics.Contains(c)).WithMessage("Invalid topic");
+        }
+
+        private static List<string> InvalidTopics => new List<string> { "announcements", "app", "docs", "settings", "stats", "mytopic-rw", "mytopic-ro", "mytopic-wo" };
+    }
+
+    public class NtfySettings : IProviderConfig
+    {
+        private static readonly NtfySettingsValidator Validator = new NtfySettingsValidator();
+
+        public NtfySettings()
+        {
+            Topics = Array.Empty<string>();
+            Priority = 3;
+        }
+
+        [FieldDefinition(0, Label = "Server Url", Type = FieldType.Url, HelpLink = "https://ntfy.sh/docs/install/", HelpText = "Leave blank to use public server (https://ntfy.sh)", Placeholder = "https://ntfy.sh")]
```

Comment:
Last comment, should we just make this the default value (public url) to avoid the majority of users typing it in?

Reply:
I'd say probably not. I had a think about that when I was creating that field.

While the service is public, it's also shared and open. So if someone sends an alert to `radarr` channel, everyone using the service who is subscribed to that channel will receive those alerts. Think of it like IRC.

To bypass that publicness, you'd need to create a channel called `radarr-qstick`.

I'd say the majority of people using this service will be self-hosted to allow the privacy aspect.

Reply:
Makes sense